code/main/media/mediaModule.brs

```diff
@@ -41,140 +41,70 @@ function _adb_MediaModule(configurationModule as object, edgeRequestQueue as obj
         ' Event data example:
         ' {
         '     clientSessionId: "xx-xxxx-xxxx",
-        '     tsObject: { ts_inISO8601: "2019-01-01T00:00:00.000Z", ts_inMillis: 1546300800000 },
+        '     tsObject: { tsInISO8601: "2019-01-01T00:00:00.000Z", tsInMillis: 1546300800000 },
         '     xdmData: { xdm: {} }
         '     configuration: { }
         ' }
         processEvent: sub(requestId as string, eventData as object)
-            mediaEventType = eventData.xdmData.xdm.eventType
-            clientSessionId = eventData.clientSessionId
-            tsObject = eventData.tsObject
-            xdmData = eventData.xdmData
+            eventType = eventData.xdmData.xdm.eventType
 
-            if mediaEventType = m._CONSTANTS.MEDIA.SESSION_START_EVENT_TYPE
-                sessionConfig = eventData.configuration
-                m._startSession(clientSessionId, sessionConfig, xdmData, tsObject)
+            if not m._hadValidConfig() then
+                _adb_logError("processEvent() - Cannot process media event (" + FormatJson(eventType) + "), missing required configuration.")
                 return
             end if
 
-            if mediaEventType <> invalid and _adb_isStringInArray(mediaEventType, m._CONSTANTS.MEDIA.EVENT_TYPES)
-                m._trackEventForSession(requestId, clientSessionId, xdmData, tsObject)
+            if not m._isValidMediaEvent(eventType) then
+                _adb_logError("processEvent() - Cannot process media event (" + FormatJson(eventType) + "), media event type (" + FormatJson(eventType) + ") is invalid.")
+                return
+            end if
+
+            sessionConfig = eventData.configuration
+            clientSessionId = eventData.clientSessionId
+
+            if eventType = m._CONSTANTS.MEDIA.EVENT_TYPE.SESSION_START
+                m._sessionManager.createSession(clientSessionId, sessionConfig, m._edgeRequestQueue)
+                m._attachMediaConfig(xdmData)
```

Comment:
Can you do this inside `sessionManager`? If not, reorder and move this to line 65.